code/datums/wires/wire_bundle_component.dm

```diff
@@ -0,0 +1,31 @@
+#define CAPACITY_PER_WIRE 10
+
+/datum/wires/wire_bundle_component
+	holder_type = /atom //Anything that can have a shell component, really.
+	randomize = TRUE
+	wire_behavior = WIRES_ALL
+
+/datum/wires/wire_bundle_component/New(atom/holder)
+	var/datum/component/shell/shell_comp = holder.GetComponent(/datum/component/shell)
+	if(!istype(shell_comp))
+		CRASH("Holder does not have a shell component!")
+	var/wire_count = clamp(round(shell_comp.capacity / CAPACITY_PER_WIRE, 1), 1, MAX_WIRE_COUNT)
+	for(var/index in 1 to wire_count)
+		wires += "Port [index]"
+	..()
+
+/datum/wires/wire_bundle_component/attach_assembly(color, obj/item/assembly/assembly)
+	var/datum/component/shell/assembly_shell_comp = assembly.GetComponent(/datum/component/shell)
+	// Recursion prevention - if the circuit is admin-only we are fine with attaching the assembly containing it
+	if(assembly_shell_comp && !assembly_shell_comp.attached_circuit?.admin_only)
+		return
```

Comment:
Seems like this should be handled on the shell component itself rather than here?

Reply:
You could send a signal and handle it on the component itself, for example. It doesn't need to be sent on the parent, sending it in this override is appropriate too.

Reply:
Could I have the assembly handle the signal?

Reply:
Sounds fine